docs/manual/src/internals/lifting_and_lowering.md

```diff
@@ -13,11 +13,55 @@ receiving side "***lifts***" that primitive type into its own language-specific
 data type.
 
 Lifting and lowering simple types such as integers is done by directly casting the
-value to and from an appropriate type. For complex types such as optionals and
-records we currently implement lifting and lowering by serializing into a byte
-buffer, but this is an implementation detail that may change in future. (See
-[ADR-0002](https://github.com/mozilla/uniffi-rs/blob/main/docs/adr/0002-serialize-complex-datatypes.md) for the reasoning
-behind this choice.)
+value to and from an appropriate type. Complex types such as optionals and
+records are lowered to a byte buffer called a `RustBuffer` internally.
+
+For example, a Python `str` is passed to Rust by lowering it to a `RustBuffer`, which is then
+lifted to a Rust `String`
+
+```mermaid
+block-beta
+  columns 2
+  block:PythonLayer
+    columns 1
+    PythonTitle["Python"]
+    PythonStr["str"]
+  end
+  block:RustLayer
+    columns 1
+    RustTitle["Rust"]
+    RustString["String"]
+  end
+  block:FFILayer:2
+    columns 3
+    space:2
+    FfiTitle["FFI"]
+    space
+    RustBuffer["RustBuffer"]
+    space
+  end
+
+  PythonStr --"Lower"--> RustBuffer
+  RustBuffer --"Lift"--> RustString
+
+classDef default stroke:transparent,fill:#fff
+classDef layer stroke:#c0c0c0,fill:#cae9ff
+classDef title stroke:transparent,fill:transparent,color:#f72585
+classDef invisible stroke:transparent,fill:transparent
+class RustLayer layer
+class PythonLayer layer
+class FFILayer layer
+class RustTitle title
+class PythonTitle title
+class FfiTitle title
+```
+
+Note: `RustBuffer` is an implementation detail that may change in future. (See
```

Comment:
I'd go further here and say "ALL of this, including the us of a RustBuffer ..." :)